src/app/static/src/app/components/modelOutput/ModelOutput.vue

```diff
@@ -309,7 +312,37 @@
                 updateSelectionsAndXAxisOrder(data, this.barchartSelections, this.barchartFlattenedXAxisFilterOptionIds, this.updateBarchartSelections)
             },
             updateComparisonPlotSelectionsAndXAxisOrder(data) {
-                updateSelectionsAndXAxisOrder(data, this.comparisonPlotSelections, this.comparisonPlotFlattenedXAxisFilterOptionIds, this.updateComparisonPlotSelections)
+                /**
+                 * If indicator changes, xaxis, disaggregate and filter
+                 * options should update with default selections
+                 */
+                const newSelection: Partial<BarchartSelections> = data
+
+                if (newSelection.indicatorId && newSelection.indicatorId !== this.comparisonPlotSelections.indicatorId) {
+                    const selections = this.comparisonPlotDefaultSelections
+                        .find(selection => selection.indicator_id === newSelection.indicatorId)
+
+                    if (selections) {
+                        const comparisonSelections = {
+                            indicatorId: selections.indicator_id,
+                            xAxisId: selections.x_axis_id,
+                            disaggregateById: selections.disaggregate_by_id,
+                            selectedFilterOptions: selections.selected_filter_options
+                        }
+                        updateSelectionsAndXAxisOrder(
+                            data,
+                            comparisonSelections,
+                            this.comparisonPlotFlattenedXAxisFilterOptionIds,
+                            this.updateComparisonPlotSelections)
+                    }
```

Comment:
Is this really necessary? `updateSelectionsAndXAxisOrder` is about re-ordering selections to conform with those in the metadata. But you are already (I think) trusting default selections to be in the right order when the first arrive (for the default indicator) so can't we also trust the order when change indicator.

Also, we don't really want to retain `data` do we, other than determining that the indicator has changed? We always want to switch back to the default selections for the new indicator, regardless of what new filter values were selected by the user for the old indicator?

In which case, I think here you'd just need to do:
`this.updateComparisonPlotSelections({payload: selections})`

Reply:
You are right! I could just mutate default selections.